### PR TITLE
fix(pfd): Fix PFD altitude indications

### PR DESCRIPTION
**Pull request by BlueberryKing**
Fixes #[issue_no]

## Summary of Changes
- Fixes current metric altitude being shown in cyan, rather than green.
- Fixes selected metric altitude showing in green rather than cyan, magenta or white.
- Fixes FCU altitude being constrained after GS capture.

Something I noticed that is not fixed by this PR is that the selected altitude indications on the PFD (both ft and meters) might briefly flicker to the FCU altitude before correctly showing a constraint altitude in magenta. I don't think this is supposed to happen.

## Screenshots (if necessary)

Before:

before-fix-no-mda

After:
after-fix-no-mda
after-fix-on-gs
after-fix-below-mda

## References

- https://youtu.be/HaizWlqYGr8
- https://youtu.be/JaTyrHFSXa4

Reference for selected metric altitude indicator in magenta.
image

Reference for selected metric indicator in cyan
image

Reference for current metric altitude below minimums
yt-pfd-metric-alt-below-minimum

## Additional context

Discord username (if different from GitHub): BlueberryKing#6641

## Testing instructions

- Load up flight, ensure metric altitude indication is shown in green.
- Make sure metric altitude indication turns amber once below MDA
- Make sure selected metric altitude indication is cyan, magenta or white, matching the color of the selected metric altitude.
- Fly an ILS approach. Once the GS is captured. Make sure that the selected altitude indication (both metric and ft) matches the FCU value and is shown in white.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/systems/instruments/src/PFD/AltitudeIndicator.tsx [hunk 1]
```diff
@@ -310,20 +341,41 @@ export class AltitudeIndicatorOfftape extends DisplayComponent<AltitudeIndicator
...
                     <RadioAltIndicator bus={this.props.bus} filteredRadioAltitude={this.props.filteredRadioAltitude} />
                     <DigitalAltitudeReadout bus={this.props.bus} />
                 </g>
             </>
         );
     }
+
+    private handleAltManagedChange() {
+        // TODO find proper logic for this (what happens when a constraint is sent by the fms but vertical mode is not managed)
+        const isSelectedModeActive = this.activeVerticalMode !== VerticalMode.OP_CLB && this.activeVerticalMode !== VerticalMode.OP_DES
```
Comment (lukecologne) [inline fbw-a32nx/src/systems/instruments/src/PFD/AltitudeIndicator.tsx:357]: Small comment: this looks like it's true if **not** in a selected mode, i.e. if we are in a managed mode, which makes sense considering the usage

File: fbw-a32nx/src/systems/instruments/src/PFD/AltitudeIndicator.tsx [hunk 2]
```diff
@@ -754,6 +776,22 @@ class MetricAltIndicator extends DisplayComponent<{ bus: EventBus }> {
...
+        } else if (this.state.altitudeColor === TargetAltitudeColor.Magenta) {
+            this.metricAltTargetText.instance.classList.remove('Cyan');
+            this.metricAltTargetText.instance.classList.remove('White');
+            this.metricAltTargetText.instance.classList.add('Magenta');
+        } else {
+            this.metricAltTargetText.instance.classList.remove('Magenta');
+            this.metricAltTargetText.instance.classList.remove('White');
+            this.metricAltTargetText.instance.classList.add('Cyan');
+        }
+    }
```
Comment (tracernz) [inline fbw-a32nx/src/systems/instruments/src/PFD/AltitudeIndicator.tsx:793]: Same for SelectedAltIndicator too
```suggestion
    private updateAltitudeColor() {
         this.metricAltTargetText.instance.classList.toggle('Cyan', this.state.altitudeColor === TargetAltitudeColor.Cyan);
         this.metricAltTargetText.instance.classList.toggle('Magenta', this.state.altitudeColor === TargetAltitudeColor.Magenta);
         this.metricAltTargetText.instance.classList.toggle('White'. this.state.altitudeColor === TargetAltitudeColor.White);
    }
```

Comment (BlueberryKing) [reply fbw-a32nx/src/systems/instruments/src/PFD/AltitudeIndicator.tsx]: Ah, I knew there had to be a better way to do this 😄. Thanks for the suggestion.

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7803
Tier of Testing  : 1
Date             : 28/02/2022

Testing Process: 
Load up flight, ensure metric altitude indication is shown in green. - **OK**
Make sure metric altitude indication turns amber once below MDA - **OK**
Make sure selected metric altitude indication is cyan, magenta or white, matching the color of the selected metric altitude. - **OK**
Fly an ILS approach. Once the GS is captured. Make sure that the selected altitude indication (both metric and ft) matches the FCU value and is shown in white. - **OK**

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Pull request merged**